src/messageschannel.hpp

```diff
@@ -61,17 +62,27 @@ class MatrixMessagesChannel : public Tp::BaseChannelTextType
 public:
     static MatrixMessagesChannelPtr create(MatrixConnection *connection, QMatrixClient::Room *room, Tp::BaseChannel *baseChannel);
 
-    QString sendMessageCallback(const Tp::MessagePartList &messageParts, uint flags, Tp::DBusError *error);
+    QString sendMessage(const Tp::MessagePartList &messageParts, uint flags, Tp::DBusError *error);
     void processMessageEvent(const QMatrixClient::RoomMessageEvent *event);
 
     void messageAcknowledgedCallback(const QString &messageId);
 
     void fetchHistory();
 
+public slots:
+    void onTypingChanged();
```

Comment:
Please group all handler together, namely `onTypingChanged` should go after `onPendingEventChanged`.
It is hard to decide where to put the methods but a 'section per method' is definitely overkilling :smile:.

Let's keep in 'public' methods that can be called from the outside without Matrix or Telepathy side confusing.
E.g. sendMessage() that accepts Telepathy MessagePartList can be called from anywhere and will do what it is means to do.

I also read the spec and found that setChatState (here and in Morse) should also set own chat state in the interface.
The interface should be changed in TelepathyQt first but we can make setChatState public with such (future) public correctness in mind.

Reply:
Tried to follow proposed changes: hidden not only non needed `public slots` section but also `protected slots`.
And also commented out not implemented `messageAcknowledged` (that should be changed  to `messageAcknowledged`)